packages/notifier/src/types-ambient.js

```diff
@@ -50,19 +62,48 @@
 
 /**
  * @template T
- * @typedef {ERef<PublicationRecord<T>>} PublicationList
- * Will be shared between machines, so it must be safe to expose. But other
- * software should consider it opaque, not depending on its internal structure.
+ * @typedef {Promise<PublicationRecord<T>>} PublicationList
+ * Will be shared between machines, so it must be safe to expose. But software
+ * outside the current package should consider it opaque, not depending on its
+ * internal structure.
  */
 
 /**
  * @template T
- * @typedef {object} Subscriber
+ * @typedef {object} EachTopic
  * @property {(publishCount?: bigint) => PublicationList<T>} subscribeAfter
- * Internally used to get the "current" SharableSubscriptionInternals
- * in order to make a subscription iterator that starts there.
- * The answer is "current" in that it was accurate at some moment between
- * when you asked and when you got the answer.
+ * Returns a promise for a "current" PublicationRecord that is later than the
+ * provided publishCount and references its successor.
+ * Used to make forward-lossless ("each") iterators.
+ */
+
+/**
+ * @template T
+ * @typedef {object} LatestTopic
+ * @property {(updateCount?: bigint | number) => Promise<UpdateRecord<T>>} getUpdateSince
+ * Returns the current or next update record as of an update count. If the
+ * `updateCount` argument is omitted or differs from the current update count,
+ * return the current record.  Otherwise, after the next state change, the
+ * promise will resolve to the then-current value of the record.  This is an
+ * attenuated form of `subscribeAfter` whose return value does not directly
+ * reference the spine of the PublicationList to ensure clients do not pin
+ * historical PublicationRecords.
```

Comment:
* @property {(updateCount?: bigint | number) => Promise<UpdateRecord<T>>} getUpdateSi
```suggestion
 * Returns a promise for an update record as of an update count.
 * If the `updateCount` argument is omitted or differs from the current update count,
 * the promise promptly resolves to the current record.
 * Otherwise, after the next state change, the promise resolves to the resulting record.
 * This is an attenuated form of `subscribeAfter` whose return value stands alone and
 * does not allow consumers to pin a chain of historical PublicationRecords.
```